src/protagonist/Engine/Ingest/IngestExecutor.cs

```diff
@@ -32,35 +36,46 @@ public class IngestExecutor
         var workers = workerBuilder.GetWorkers(asset);
         
         var context = new IngestionContext(asset);
+        var overallStatus = IngestResultStatus.Unknown;
 
         if (!assetIngestorSizeCheck.CustomerHasNoStorageCheck(asset.Customer))
         {
             var preIngestionAssetSize = await assetRepository.GetImageSize(asset.Id, cancellationToken);
             context.WithPreIngestionAssetSize(preIngestionAssetSize);
+            
+            var counts = await storageRepository.GetStorageMetrics(asset.Customer, cancellationToken);

```

Comment:
Can this check go before `var preIngestionAssetSize = await assetRepository.GetImageSize(...)`? If so it could save a DB hit to get `ImageStorage` record in the event of storage being exceeded.